lib/src/main/java/org/esbtools/lightbluenotificationhook/NotificationHook.java

```diff
@@ -88,7 +96,9 @@ public void processHook(EntityMetadata entityMetadata,
         Mediator mediator = tryGetMediator();
 
         Projector watchProjector = Projector.getInstance(config.watchProjection(), entityMetadata);
-        Projector includeProjector = Projector.getInstance(config.includeProjection(), entityMetadata);
+        Projection includeProjection=addArrayIdentities(config.includeProjection(),entityMetadata);
+        Projector includeProjector = Projector.getInstance(includeProjection, entityMetadata);
```

Comment:
Not sure, but I think we may have wanted to modify the watch projection instead of the include projection.

Here is a test case which demonstrates the bug I was running into. In this PR branch it does not pass (though double check me to make sure I don't have something wrong):

```java
    @Test
    public void shouldOnlyIncludeWhatChangedOfArrayElementIfElementIdentityNotWatched()
            throws Exception {
        EntityMetadata md = getMd("usermd.json");
        JsonNode pre = loadJsonNode("userdata.json");
        JsonNode post = loadJsonNode("userdata.json");
        JsonDoc.modify(post, new Path("sites.0.streetAddress.city"), JsonNodeFactory.instance.textNode("blah"), true);

        List<HookDoc> docs= new ArrayList<>();
        HookDoc doc = new HookDoc(md, new JsonDoc(pre), new JsonDoc(post), CRUDOperation.UPDATE, "me");
        docs.add(doc);

        HookConfiguration cfg = new NotificationHookConfiguration(
                projection("{'field':'sites.*.streetAddress.city','recursive':1}"),
                null,
                false);

        hook.processHook(md, cfg, docs);
        Assert.assertNotNull(insertCapturingMediator.capturedInsert);
        JsonNode data=insertCapturingMediator.capturedInsert.getEntityData();

        Truth.assertThat(
                Iterables.transform(data.get("changedPaths"), toTextValue()))
                .containsExactly("sites.0.streetAddress.city");
    }
```